pkg/apis/core/v1alpha1/types.go

```diff
@@ -23,9 +23,15 @@ import (
 	conditions "github.com/openshift/custom-resource-status/conditions/v1"
 )
 
-// DataVolume provides a representation of our data volume
+// DataVolume is an abstraction on top of PersistentVolumeClaims to allow easy population of those PersistentVolumeClaims with relation to VirtualMachines
 // +genclient
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:shortName=dv;dvs,categories=all
+// +kubebuilder:printcolumn:name="Phase",type="string",JSONPath=".status.phase",description="The phase the data volume is in"
+// +kubebuilder:printcolumn:name="Progress",type="string",JSONPath=".status.progress",description="Transfer progress in percentage if known, N/A otherwise"
+// +kubebuilder:printcolumn:name="Restarts",type="integer",JSONPath=".status.restartCount",description="The number of times the transfer has been restarted."
+// +kubebuilder:printcolumn:name="Age",type="date",JSONPath=".metadata.creationTimestamp"
```

Comment:
So the tool is part of the kubebuilder project, reference can be found here: https://book.kubebuilder.io/reference/markers/crd.html. One of the tools in kubebuilder is called controller-gen, it will generate CRD yamls based on the types.go files. So the thought is, I can now generate proper validation schemas from types.go, as well as generate proper validation  schemas from our code schema as yaml, and I can compare the two to prevent drift from happening (the part that I haven't added to this PR yet)

Reply:
thanks for info, so it is a standard tool for k8s apis and the hint is in +kubebuilder, so I think it does not need more explanation, I can check the docs